Improvements and fixes

kpeu3u opened the pull request.

kpeu3u committed:
change codigniter4/shiled version to dev-develop

kpeu3u committed:
Fix config namespaces in accordance with Codeigniter 4.2.6

kpeu3u committed:
Apply Fix TypeError when Saving a User from PR - https://github.com/lonnieezell/Bonfire2/pull/194
Fix some return types

kpeu3u committed:
Refactor of fix TypeError when Saving a User

kpeu3u committed:
Update Auth Config file

kpeu3u committed:
Fix actions when saving user settings
Added actions for Email2FA and EmailActivator

kpeu3u committed:
Remove unneeded methods from auth actions Email2FA and EmailActivator

kpeu3u committed:
include zxcvbn.js file and fix path to zxcvbn.js using asset_link

kpeu3u committed:
include zxcvbn.js file and fix path to zxcvbn.js using asset_link

kpeu3u committed:
include zxcvbn.js file and fix path to zxcvbn.js using asset_link

kpeu3u committed:
update zxcvbn.js to v4.4.2

kpeu3u committed:
change tests namespaces

kpeu3u committed:
remove route namespaces
define route variable

kpeu3u committed:
remove return type

kpeu3u committed:
run cs-fix

kpeu3u committed:
remove unneeded file spark as it is a part of framework

kpeu3u committed:
composer update

kpeu3u committed:
fix tests

kpeu3u committed:
Merge branch 'lonnieezell_develop' into develop

# Conflicts:
#	composer.lock
#	src/Users/Controllers/UserController.php
#	src/Users/Views/security.php
#	src/Views/Auth/register.php

lonnieezell commented on src/Auth/Authentication/Actions/Email2FA.php:
```diff
@@ -0,0 +1,144 @@
+<?php
```
I don't think there's a reason to port this from Shield. It just makes a maintenance issue to keep it up to date to changes from Shield. Was there a valid reason to bring it forward?

kpeu3u replied:
The reason to port this is that i include Themeable trait and made methods to use Auth config items to render templates

lonnieezell replied:
Ah, gotcha. It looks like Shield is using the `settings()` command instead of `config`. 

I am really uncomfortable duplicating all of the code here. I think a better solution is to submit a PR to Shield for the actions to use a `renderView()` method in the controllers so that all that needs to be extended is that one method and all of the other logic stays the same. 

Please remove those changes from this PR and we can get the rest merged, and then work on the other changes.

lonnieezell replied:
Thanks for updating that. Will merge.

lonnieezell commented on src/Auth/Authentication/Actions/EmailActivator.php:
```diff
@@ -0,0 +1,122 @@
+<?php
```
I don't think there's a reason to port this from Shield. It just makes a maintenance issue to keep it up to date to changes from Shield. Was there a valid reason to bring it forward?

kpeu3u replied:
The reason to port this is that i include Themeable trait and made methods to use Auth config items to render templates

lonnieezell commented on src/Commands/Install/Publisher.php:
```diff
@@ -58,12 +58,11 @@ public function publishClass(string $className): void
 
         $replace = [
-            $namespace                           => $newNamespace,
+            $namespace                           => 'Config',
```
Has this been tested? It seems like would be missing something and I'm pretty sure the publisher does its job correctly currently?

kpeu3u replied:
This change is because of Fatal error: Cannot declare class Config\Auth\Auth, because the name is already in use in /app/Config/Auth.php on line 0
since after publish namespace becomes: namespace Config\Auth;

lonnieezell commented on src/Users/Views/user_settings.php:
```diff
@@ -34,8 +34,8 @@
                             <input class="form-check-input" type="checkbox" name="emailActivation"
-                                   value='Shield\Authentication\Actions\EmailActivate' id="email-activation"
-                                <?php if (old('emailActivation', setting('Auth.actions')['register']) === 'Shield\Authentication\Actions\EmailActivate') : ?>
+                                   value='Bonfire\Auth\Authentication\Actions\EmailActivator' id="email-activation"
```
As per comments above, I'm not sure this is really needed and would require a maintenance headache.

kpeu3u replied:
this is related to Themeable trait end setting templates

lonnieezell commented on src/Users/Views/user_settings.php:
```diff
@@ -34,8 +34,8 @@
-                                   value='Shield\Authentication\Actions\EmailActivate' id="email-activation"
-                                <?php if (old('emailActivation', setting('Auth.actions')['register']) === 'Shield\Authentication\Actions\EmailActivate') : ?>
+                                   value='Bonfire\Auth\Authentication\Actions\EmailActivator' id="email-activation"
+                                <?php if (old('emailActivation', setting('Auth.actions')['register']) === 'Bonfire\Auth\Authentication\Actions\EmailActivator') : ?>
```
As per comments above, I'm not sure this is really needed and would require a maintenance headache.

kpeu3u replied:
this is related to Themeable trait end setting templates

lonnieezell commented on src/Users/Views/user_settings.php:
```diff
@@ -120,8 +120,8 @@
                             <input class="form-check-input" type="checkbox" name="email2FA"
-                                   value="Shield\Authentication\Actions\Email2FA" id="email-2fa"
-                                <?php if (old('email2FA', setting('Auth.actions')['login']) === 'Shield\Authentication\Actions\Email2FA') : ?>
+                                   value="Bonfire\Auth\Authentication\Actions\Email2FA" id="email-2fa"
```
As per comments above, I'm not sure this is really needed and would require a maintenance headache.

kpeu3u replied:
this is related to Themeable trait end setting templates

lonnieezell commented on src/Users/Views/user_settings.php:
```diff
@@ -120,8 +120,8 @@
-                                   value="Shield\Authentication\Actions\Email2FA" id="email-2fa"
-                                <?php if (old('email2FA', setting('Auth.actions')['login']) === 'Shield\Authentication\Actions\Email2FA') : ?>
+                                   value="Bonfire\Auth\Authentication\Actions\Email2FA" id="email-2fa"
+                                <?php if (old('email2FA', setting('Auth.actions')['login']) === 'Bonfire\Auth\Authentication\Actions\Email2FA') : ?>
```
As per comments above, I'm not sure this is really needed and would require a maintenance headache.

kpeu3u replied:
this is related to Themeable trait end setting templates

lonnieezell commented on src/Widgets/Controllers/WidgetsSettingsController.php:
```diff
@@ -82,7 +82,7 @@ public function save(): \CodeIgniter\HTTP\RedirectResponse
             default:
                 $this->saveWidgetSettings();
-            // dd($this->request->getVar('widget'));
+                // dd($this->request->getVar('widget'));
```
We can delete this line.

```suggestion
```

kpeu3u replied:
This line has been deleted, i just missed it.

kpeu3u committed:
Merge branch 'lonnieezell:develop' into develop

kpeu3u committed:
remove unneeded commented debug line

kpeu3u committed:
Merge remote-tracking branch 'origin/develop' into develop

kpeu3u committed:
run cs-fix

kpeu3u committed:
update shield and get rid of unit test errors 1 and 2
since in the last commit shild have merged my PR that fix UserModel
--------------------

There were 3 errors:

1) Tests\Auth\AdminAccessTest::testAdminsCanAccess
Error: Call to a member function set() on null

/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Models/UserModel.php:347
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Authentication/Authenticators/Session.php:880
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Filters/SessionAuth.php:49
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Filters/Filters.php:173
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:455
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:351
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:187
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:219
/Users/kpeu3u/GitProjects/Forks/Bonfire2/tests/Auth/AdminAccessTest.php:34
phpvfscomposer:///Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/phpunit/phpunit/phpunit:97
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/bin/phpunit:115

2) Tests\Auth\AdminAccessTest::testCannotAccessAdminWithoutPermission
Error: Call to a member function set() on null

/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Models/UserModel.php:347
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Authentication/Authenticators/Session.php:880
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Filters/SessionAuth.php:49
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Filters/Filters.php:173
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:455
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:351
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:187
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:219
/Users/kpeu3u/GitProjects/Forks/Bonfire2/tests/Auth/AdminAccessTest.php:51
phpvfscomposer:///Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/phpunit/phpunit/phpunit:97
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/bin/phpunit:115

3) Tests\Auth\AdminAccessTest::testCannotViewNavItemsWithoutPermission
Error: Call to a member function set() on null

/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Models/UserModel.php:347
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Authentication/Authenticators/Session.php:880
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/shield/src/Filters/SessionAuth.php:49
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Filters/Filters.php:173
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:455
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/CodeIgniter.php:351
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:187
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/codeigniter4/framework/system/Test/FeatureTestTrait.php:219
/Users/kpeu3u/GitProjects/Forks/Bonfire2/tests/Auth/AdminAccessTest.php:70
phpvfscomposer:///Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/phpunit/phpunit/phpunit:97
/Users/kpeu3u/GitProjects/Forks/Bonfire2/vendor/bin/phpunit:115

ERRORS!
Tests: 105, Assertions: 259, Errors: 3.

kpeu3u committed:
remove Authentication\Actions and revert user settings view
